### PR TITLE
Downstream url documentation

### LINKED ISSUE #3039: Add better documentation for downstream_url
This might be subjective and is solely based on my own experience. The docs for `-frontend.downstream-url` do not sufficiently explain what will happen when using it:
```yaml
# URL of downstream Prometheus.
# CLI flag: -frontend.downstream-url
[downstream_url: <string> | default = ""]
```

My impression was that this had something to do with `generator_url` in alerts etc. (don't ask me how I arrived at that conclusion though). Nevertheless, I pointed it at my actual prometheus instance, which resulted in the query-frontend returning 404 errors for all endpoints, and I couldn't for the life of me figure out why. Only when I integrated Jaeger and got some tracing up and running did I see that the requests ended up at Prometheus.

tbh I still don't understand what this is *for*. [I can see what it does](https://github.com/cortexproject/cortex/blob/9c6b2293ef1bc654ccdf16177f803546cf2715da/pkg/querier/frontend/frontend.go#L108-L111), but I don't understand the motivation.

### PR DESCRIPTION
**Pull request by srijan55**
**What this PR does**:
Adds some documentation about downstream_url config in Frontend.
**Which issue(s) this PR fixes**:
Fixes #3039

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -210,6 +210,8 @@ Flow of the query in the system when using query-frontend:
 3) Querier picks up the query, and executes it.
 4) Querier sends result back to query-frontend, which then forwards it to the client.
 
+Query frontend can also be used with any Prometheus-API compatible service. In this mode Cortex can be used as an query accelerator with it's caching and splitting features on other prometheus query engines like Thanos Querier or your own Prometheus server.  Query frontend needs to be configured with downstream url address(via the `-frontend.downstream-url` CLI flag), which is endpoint of the prometheus server intended to be connected with Cortex.
```
Comment (andsens) [inline docs/architecture.md:213]: > which is endpoint of the prometheus server

"which is *the* endpoint...

File: docs/configuration/prometheus-frontend.md [hunk 2]
```diff
@@ -66,6 +66,6 @@ frontend:
   log_queries_longer_than: 1s
   compress_responses: true
 
-  # The Prometheus URL to which the query-frontend should connect to.
+  # The query endpoint URL of Prometheus-API compatible service to which the query-frontend should connect to.
```
Comment (andsens) [inline docs/configuration/prometheus-frontend.md:69]: > of Prometheus-API

"of *a* Prometheus-API"

File: docs/configuration/prometheus-frontend.yml [hunk 3]
```diff
@@ -54,5 +53,5 @@ frontend:
   log_queries_longer_than: 1s
   compress_responses: true
 
-  # The Prometheus URL to which the query-frontend should connect to.
+  # The query endpoint URL of Prometheus-API compatible service to which the query-frontend should connect to.
```
Comment (andsens) [inline docs/configuration/prometheus-frontend.yml:56]: > of Prometheus-API

"of *a* Prometheus-API"

**Review by andsens — CHANGES_REQUESTED** [review]
Awesome. This explains very succinctly what the parameter is for.

**Comment by srijan55** [comment]
> Awesome. This explains very succinctly what the parameter is for.

Thanks. I made an unsigned commit and to resolve it did something wrong ... will clean it up or may be abandon this and create a new one.

**Comment by srijan55** [comment]
@andsens  have a look again when you have a moment. Also suggest someone who can be second reviewer and also help with the merge if everything is fine

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,7 @@
 * [ENHANCEMENT] Added new ring related config `-distributor.excluded-zones` when set this will exclude the comma-separated zones from the ring, default is "". #4539
 * [ENHANCEMENT] Upgraded Docker base images to `alpine:3.14`. #4514
 * [ENHANCEMENT] Updated Prometheus to latest. Includes changes from prometheus#9239, adding 15 new functions. Multiple TSDB bugfixes prometheus#9438 & prometheus#9381. #4524
+* [BUGFIX] Updated documentation with more details about `-frontend.downstream_url`. #3039
```
Comment (alvinlin123) [inline CHANGELOG.md:10]: I don't think we typically up a CHANGELOG item for doc update (@bboreham correct me if I'm wrong). But if we do, I would say this change is more of an `ENHANCEMENT` than `BUGFIX` :-)

**Review by alvinlin123 — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (friedrich-at-adobe) [inline CHANGELOG.md:10]:
```suggestion
```

**Pull request merged**